frontend/src/components/table/TableHeader.js

```diff
@@ -145,4 +145,23 @@ TableHeader.propTypes = {
   cols: PropTypes.any,
   indentSupported: PropTypes.any,
   setActiveSort: PropTypes.func,
+  headersFields: PropTypes.object,
+  setActiveSortFields: PropTypes.func,
 };
+
+const mapStateToProps = (state, ownProps) => {
+  const { viewId, windowType: windowId, docId, tabId } = ownProps;
+  const tableId = getTableId({ windowId, viewId, docId, tabId });
+
+  return {
+    headersFields:
+      state.tables && state.tables.headers ? state.tables.headers[tableId] : {},
+  };
+};
+
+export default connect(
+  mapStateToProps,
+  {
+    setActiveSortFields,
+  }
+)(TableHeader);
```

Comment:
Why not add a named export of a component disconnected from the store ? This way you won't have to do the whole shenanigans in the tests and use what we've had instead. Then the reducers/actions can be tested separately. We know, that `mapStateToProps` works as redux has a pretty good test coverage.

Reply:
Not sure what you mean in here. Don't we want the components to be independent of each other ?

Reply:
I believe you were thinking on importing directly the component disconnected from the file. Done that and replaced the test with old one. With some small adapts